### ISSUE TITLE
PMU Question Flow

### ISSUE THREAD

**Issue by bamboochow** [whynotearth/BrowTricks#93]
# Design File
For design reference, please see files in [Figma](https://www.figma.com/file/m8JgvnHpgPmGq8Bm2iAf1a/Brow-Tricks?node-id=709%3A10179)

**You can find the designs for this issue in the "Tenant CMS" Page of Figma**

Remained things:  
- [ ] Animations
- [x] Validation
- [ ] Preview answers

# PMU Question Flow 
This is a step by step flow of questions for a PMU consent for that can be generated into a PDF at the end of the flow 
![image](https://user-images.githubusercontent.com/14948326/85380153-c84cf600-b566-11ea-9828-d89af20d9ecd.png)

The flow starts with a page that displays options for completing the form in person, or sent to client's preferred contact information 
![image](https://user-images.githubusercontent.com/14948326/85376496-fda31500-b561-11ea-9d90-206228f1cbdd.png)

# Get Started 
"Complete in person" redirects to the "Get started" page within the app 

Client's preferred contact will either send them an email (included in separate issue) or a text with a link to start at the "get started" page 

![image](https://user-images.githubusercontent.com/14948326/85377142-d9940380-b562-11ea-9f0f-5c7774e824f4.png)

# Question Flow 

The content of the question flow can be copied directly from the Figmadesigns
For design reference, please see files in [Figma](https://www.figma.com/file/m8JgvnHpgPmGq8Bm2iAf1a/Brow-Tricks?node-id=709%3A10179)

**You can find the designs for this issue in the "Tenant CMS" Page of Figma**

Each step includes: 
1. stepper navigation 
2. content

Starting at step 2, the form will act like this Typeform example where content moves up when you press continue, but you're always allowed to scroll up and review past questions (scrolling up will act identical to "back") https://www.typeform.com/templates/t/client-consultation-form-template/

## Stepper Navigation 
The stepper navigation is identical to this 
https://github.com/whynotearth/BrowTricks/issues/64

The code for a similar component from foodouken is here: 
https://github.com/whynotearth/foodouken.com/blob/staging/src/components/BaseStepperTopBar.vue
https://github.com/whynotearth/foodouken.com/blob/staging/src/components/BaseStepperBottomBar.vue

## Content 
**Step 1: Signature Creation** 
This step has 2 text fields, one for signature, one for initials, with a preview for each below in script text 
Code for text fields: https://github.com/whynotearth/BrowTricks/blob/master/src/components/BaseInputText.vue

![image](https://user-images.githubusercontent.com/14948326/85380200-d7cc3f00-b566-11ea-9745-b1990ffce048.png)

**Step 2-3: Disclosure** 
These are read-only text that explains the procedure and the agreement
![image](https://user-images.githubusercontent.com/14948326/85380225-e155a700-b566-11ea-90b6-7c3e77e39c91.png)

**Step 4: Release** 
This questions authorizes photo permission. 
![image](https://user-images.githubusercontent.com/14948326/85380311-fcc0b200-b566-11ea-954c-3903027d9ec8.png)

**Step 5-16: Medical Disclaimer** 
These are a series of medical disclaimer questions. 
For yes/no questions, selecting "Yes" automatically proceeds to the next step, while selecting "no" AFTER STEP 8 disables the "continue" button.
However, If they answer "no" to step 5, immediately skip to step 9.
![image](https://user-images.githubusercontent.com/14948326/85380382-13670900-b567-11ea-9666-bfc551dbb89d.png)
![image](https://user-images.githubusercontent.com/14948326/85380413-1e219e00-b567-11ea-94b1-374326786a0b.png)

**Step 17-19: Review and Sign** THIS IS THE DEFAULT FLOW FOR NO ADDITION QUESTION, SEE COMMENT BELOW
These are questions confirming user have read the content and post procedure instructions. On step 19, a preview of the PDF is generated with full signatures (IN SEPARATE ISSUE). Once they press finish, they are sent a copy of their PDF agreement. 
![image](https://user-images.githubusercontent.com/14948326/85380449-27126f80-b567-11ea-8710-44160b2d2498.png)

**Comment by bamboochow** [comment]
@ctyar We are adding custom questions into our current PMU question flow. 

Step 17-???: Review and Sign
These are questions confirming user have read the content and post procedure instructions. On the last step, a preview of the PDF is generated with full signatures (IN SEPARATE ISSUE). Once they press finish, they are sent a copy of their PDF agreement.

after step 18, we are adding custom question pages. The first page is a disclaimer that these questions are specific to the tenant, and after that step, each additional question is displayed in a text input format like the second page below, based on the questions added in the PMU form setup #119. The stepper navigation will display each step (19, 20, 21 etc) as each question progresses. 

These additional questions will be displayed on a third page of the created PDF 

![image](https://user-images.githubusercontent.com/14948326/86999925-ff194200-c1dd-11ea-9f5d-f0892d2ba4f5.png)

**Comment by ctyar** [comment]
@bamboochow Do we have a design/task for the page that admin adds these questions?

**Comment by bamboochow** [comment]
@ctyar Yes 
This is the issue https://github.com/whynotearth/BrowTricks/issues/119

**Comment by bamboochow** [comment]
@mort3za 
Updated PMU Questions pages to checkboxes instead of radio buttons 
![image](https://user-images.githubusercontent.com/14948326/88634646-da2f3500-d0e0-11ea-9e84-47c3afda49e1.png)
![image](https://user-images.githubusercontent.com/14948326/88634675-e5826080-d0e0-11ea-8d03-04740b6e6889.png)

**Comment by bamboochow** [comment]
# Signature Page 
Floating label in text field should have color matching background instead of the little white squares 
![unknown](https://user-images.githubusercontent.com/14948326/88813490-2f536f80-d1e3-11ea-87e7-3588d3f99363.png)

# Text Sizing 
Yes/No Text sizing seems bigger than body text? Should be the same (mobile body text) 
![unknown (1)](https://user-images.githubusercontent.com/14948326/88813556-472af380-d1e3-11ea-9f94-f2a3b7d6db18.png)

# Remove question 
Client asked to remove this question 
![image](https://user-images.githubusercontent.com/14948326/88813892-ab4db780-d1e3-11ea-8e5f-ff49439a3530.png)

# Mobile Margins 
Content margins for mobile is way too tight, design shows a 16 pixel margin for mobile 
![unknown (2)](https://user-images.githubusercontent.com/14948326/88814018-c7e9ef80-d1e3-11ea-9c9b-3d62c2bd40a2.png)

**Comment by paulchrisluke** [comment]
@mort3za  if we can't pre-fill the hello sign document then there is no need for our own UI for question answering. In that case we can skip our own UI and just hyperlink straight to hello sign. The problem is how do we add custom questions if they add them? @ctyar @StrangeWill 

https://browtricks.netlify.app/ Go to log in, create profile, create client, click client, click pmu form button
